src/java.xml/share/classes/com/sun/org/apache/xerces/internal/impl/xs/traversers/XSAttributeChecker.java

```diff
@@ -1816,7 +1816,7 @@ OneAttr get(String key) {
 class LargeContainer extends Container {
     Map<String, OneAttr> items;
     LargeContainer(int size) {
-        items = new HashMap<>(size*2+1);
+        items = HashMap.newHashMap(size*2+1);
```

Comment:
I'm wondering if we should change this to just `newHashMap(size)` since it looks like this container is intended to hold up to `size` items. @JoeWang-Java ? I suspect the `size*2+1` was a failed attempt at allocating a HashMap of the correct capacity for `size` mappings.

Reply:
> I'm wondering if we should change this to just `newHashMap(size)` since it looks like this container is intended to hold up to `size` items. @JoeWang-Java ? I suspect the `size*2+1` was a failed attempt at allocating a HashMap of the correct capacity for `size` mappings.

I looked the codes and don't think so..
If I'm wrong, I'm glad to fix.

Reply:
Stuart's right, I looked at the code, it's as you said a failed attempt, "size" would be good. So HashMap.newHashMap(size) would actually be a small improvement.

It's an interesting impl the way it used HashMap, but it's 20 year code.

Reply:
@JoeWang-Java @stuart-marks got it. done.